packages/compile-common/src/index.ts

```diff
@@ -1,2 +1,62 @@
 import { Profiler } from "./profiler";
 export { Profiler };
+
+type Compilation = {
+  sourceIndexes: string[];
+  contracts: CompiledContract[];
+  compiler: {
+    name: string;
+    version: string;
+  };
+};
+
+type CompilerResult = Compilation[];
+
+type CompiledContract = {
+  contractName: string;
+  contract_name: string;
+  sourcePath: string;
+  source: string;
+  sourceMap: string;
+  deployedSourceMap: string;
+  legacyAST: object;
+  ast: object;
+  abi: object[];
+  metadata: string;
+  bytecode: string;
+  deployedBytecode: string;
+  compiler: {
+    name: string;
+    version: string;
+  };
+  devdoc: object;
+  userdoc: object;
+  immutableReferences: object;
+};
+
+type ContractObject = {
```

Comment:
You can just import this:

```typescript
import { ContractObject } from "@truffle/contract-schema/spec";
```

(This might require adding that package as a `dependency`, note!)

Reply:
How does that work here with TS? Does that mean a dev dep?